### ISSUE TITLE
.Net Core support

### ISSUE THREAD

**Issue by ThirtySomething** [Tinkerforge/weather-station#3]
I am currently working on a project to read data from the weather station (currently in progress) and save/display it later (planned). So far I'm working with C#, and that works perfectly. The installation of the NuGet package for Tinkerforge was a breeze. Thank you for the great integration and support.

Are there any plans to support Net Core? I would like to run the program later on a Raspberry Pi. Net Core would offer itself for this purpose: development on a Windows PC and later on run the program on a simple device.

**Comment by photron** [comment]
Are you referring to this warning?

> warning NU1701: Package 'Tinkerforge 2.1.16' was restored using '.NETFramework,Version=v4.6.1' instead of the project target framework '.NETCoreApp,Version=v2.0'. This package may not be fully compatible with your project.

If not, then what do you mean with .NET Core support? Because the C# API bindings are already working with .NET Core. Only the NuGet package doesn't contain a dedicated version for the .NET Core framework at the moment.

**Comment by ThirtySomething** [comment]
@photron Thanks for the note - I'm currently updating my C# capabilities and the project is working with C# on Windows. At the moment I have no experience with C# on Linux or .Net Core.

You're right, my first try on Linux triggered the warning you mentioned and the program itself didn't work like on Windows. However, this is because the loader contains Reflection Code for the plugins that is not valid for .Net Core. I'll have to make adjustments to fix this.

I'll report back when I'm done. Based on your answer I assume that the Tinkerforge library/the NuGet package will also work on .Net Core. To be precise - with .Net Core I'm thinking about the [framework ](https://www.microsoft.com/net/download/windows) from Microsoft, available for Windows and also for Linux.

**Comment by photron** [comment]
Yes, the Tinkerforge library works with .NET Core. But the NuGet package didn't contain a dedicated version of the library for .NET Core. But the .NET Core tools will just pick the version for .NET 4.6.1 instead and that works as well, apart from the mentioned warning.

I've updated the NuGet package and added a dedicated version of the library for .NET Core. This fixes the warning in my tests.

**Comment by ThirtySomething** [comment]
@photron I've checked my project. now it's woring on Linux, too. I've made some mistakes:
- The code for loading an assembly is different on .Net Core to my solution. I've fixed this.
- Then I've got a "connection refused" error - I've forgot to install the brick daemon.

Now it's working - there is still an issue, but this is not related to the TinkerForge package. While running the program from command line, I'm getting no values. Running the program in VSCode while debugging, I'm getting values. I'm checking my implementation to fix this.

Thank you for fixing the waring with the version number +1. During my tests I the warning no longer appeared.
